e2e/testcases/reconciler_manager_test.go

```diff
@@ -754,6 +754,10 @@ func hasTolerations(tolerations []corev1.Toleration) testpredicates.Predicate {
 		if !ok {
 			return testpredicates.WrongTypeErr(d, &appsv1.Deployment{})
 		}
+		if len(d.Spec.Template.Spec.Tolerations) != len(tolerations) {
```

Comment:
I would add some commentary as to why the length of tolerations is actually important.

Reply:
nit: I find it fairly self-evident, but added a comment